llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -3209,10 +3209,12 @@ class LLVM_ABI TargetLoweringBase {
   /// \p Shuffles is the shufflevector list to DE-interleave the loaded vector.
   /// \p Indices is the corresponding indices for each shufflevector.
   /// \p Factor is the interleave factor.
+  /// \p MaskFactor is the interleave factor that considers mask, which can
+  /// reduce the original factor.
   virtual bool lowerInterleavedLoad(Instruction *Load, Value *Mask,
                                     ArrayRef<ShuffleVectorInst *> Shuffles,
-                                    ArrayRef<unsigned> Indices,
-                                    unsigned Factor) const {
+                                    ArrayRef<unsigned> Indices, unsigned Factor,
+                                    unsigned MaskFactor) const {
     return false;
```

Comment:
Would it be a bit easier for the targets if we instead passed the stride in bytes in? That way they wouldn't have to worry about the difference between the MaskFactor and Factor. 

Targets that don't support strided interleaved loads would check that `Stride == DL.getTypeStoreSize(VTy->getElementType())`

Reply:
> Targets that don't support strided interleaved loads would check that `Stride == DL.getTypeStoreSize(VTy->getElementType())`

I believe stride is relative to the current start address, so in the case of skipping fields, the stride will always be `Factor * DL.getTypeStoreSize(VTy->getElementType())` regardless of how many fields you wanna skip.
But I guess my more high-level question would be: for those targets that don't support strided interleaved loads, what is the benefit of replacing a check between Factor and MaskFactor with another check on Stride ?

Reply:
> add a separate "lowerStridedInterleaved" TTI hook

(I believe you meant TLI hooks) Yeah I'm also worried about the fact that it will double the number of hooks, as all four of the them could have a strided version.